kratos/processes/find_global_nodal_entity_neighbours_process.h

```diff
@@ -0,0 +1,177 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//                   Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_FIND_GLOBAL_NODAL_ENTITY_NEIGHBOURS_PROCESS_H_INCLUDED )
+#define  KRATOS_FIND_GLOBAL_NODAL_ENTITY_NEIGHBOURS_PROCESS_H_INCLUDED
+
+// System includes
+#include <string>
+#include <iostream>
+#include <unordered_map>
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "processes/process.h"
+#include "includes/model_part.h"
+#include "includes/global_pointer_variables.h"
+#include "utilities/communication_coloring_utilities.h"
+#include "utilities/pointer_communicator.h"
+#include "includes/mpi_serializer.h"
+#include "utilities/global_pointer_utilities.h"
+
+namespace Kratos
+{
+
+///@name Kratos Globals
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Short class definition.
+/** Detail class definition.
+*/
+
+template<class TContainerType>
+class KRATOS_API(KRATOS_CORE) FindGlobalNodalEntityNeighboursProcess : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    using IndexType = std::size_t;
+
+    using NodeType = ModelPart::NodeType;
+
+    using EntityType = typename TContainerType::value_type;
+
+    using GlobalEntityPointersVectorType = GlobalPointersVector<EntityType>;
+
+    //contains id vs vector_of_neighbours
+    using NeighbourMapType = std::unordered_map<int, GlobalPointersVector<EntityType>>;
+
+    using NonLocalMapType =  std::unordered_map<int, NeighbourMapType>;
+
+    /// Pointer definition of FindGlobalNodalEntityNeighboursProcess
+    KRATOS_CLASS_POINTER_DEFINITION(FindGlobalNodalEntityNeighboursProcess);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+    FindGlobalNodalEntityNeighboursProcess(
+        ModelPart& rModelPart);
+
+    FindGlobalNodalEntityNeighboursProcess(
+        ModelPart& rModelPart,
+        const Variable<GlobalEntityPointersVectorType>& rOutputVariable);
+
+    /// Destructor.
+    ~FindGlobalNodalEntityNeighboursProcess() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    FindGlobalNodalEntityNeighboursProcess<EntityType>& operator=(FindGlobalNodalEntityNeighboursProcess<EntityType> const& rOther) = delete;
+
+    void operator()()
+    {
+        Execute();
+    }
+
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void Execute() override;
+
+    void ClearNeighbours();

```

Comment:
```suggestion
    void Clear() override;
```
This is the one in the base `Process` API. Then we move the implementation inside the `ClearNeighbours` to `Clear`. I'd keep the `ClearNeighbours` for a while to issue a deprecation warning.

Reply:
done :)